### PR TITLE
Update validation and reduction. 

### LINKED ISSUE #212: Provide optional validation
**Is your feature request related to a problem? Please describe.**
Currently, there is no way to skip the input validation (`_validate_input`) which is useful during prototyping and debugging but slows down the implementations. It could be interesting to add the option to skip the validation.

**Describe the solution you'd like**
With the flag `-O`, the Python interpreter doesn't consider `assert` statements. In fact, `-O` sets the global variable `__debug__` to `False`. The following could be added to the start of `_validate_input` to shortcut all the validation when `-O` is used:

```python
if not __debug__:
    return
```

**Describe alternatives you've considered**
Alternatively, a global variable `_VALIDATION`, a function `_is_validation_enabled()` and a function (or a context) `_set_validation_enabled()` could be provided.

```python
def _is_validation_enabled() -> bool:
    return _VALIDATION

def _set_validation_enabled(mode: bool) -> None:
    _VALIDATION = mode
```

### PR DESCRIPTION
**Pull request by zakajd**
Closes #212 
Another semantic update =)

## Proposed Changes
  - Add `_reduction` function to unify reduction process.
  - Merge `_validate_features` and `_validate_input` into one function and simplify it's interface
  - Unify docsting descriptions for reduction and data_range. 

With this PR we make requirements on input tensors more strict. 
- Drop support for 2D and 3D tensors
- Drop support for scale weights as lists

- [x] Add section to README with description of using `-o` flag after #232 is merged.
image

### CODE REVIEW

**Comment by snk4tr** [comment]
@zakajd are you still working on this one?

**Comment by zakajd** [comment]
@snk4tr  Yeah, I'm waiting for the documentation PR to be merged so that I can update it with -o flag usage example.

**Comment by zakajd** [comment]
@snk4tr This one is ready for merge. 
Main change is in `piq/utils/common.py`, all other 30+ files are mostly test fixes and one-line changes to adjust for new reduction and validation.

File: piq/ssim.py [hunk 1]
```diff
@@ -275,24 +255,16 @@ class MultiScaleSSIMLoss(_Loss):
 
     Args:
         kernel_size: By default, the mean and covariance of a pixel is obtained
-            by convolution with given filter_size.
+            by convolution with given filter_size. Must be an odd value.
```
Comment (snk4tr) [inline piq/ssim.py:258]: Same about odd value here

**Review by snk4tr — DISMISSED** [review]
Nice one, @zakajd. 
Let me add a few tiny fixes and merge.

**Review by snk4tr — APPROVED** [review]

**Pull request merged**